Lib/test/test_math.py

```diff
@@ -475,6 +476,17 @@ def testCopysign(self):
         # similarly, copysign(2., NAN) could be 2. or -2.
         self.assertEqual(abs(math.copysign(2., NAN)), 2.)
 
+    def test_signbit(self):
+        self.assertRaises(TypeError, math.signbit)
+        self.assertRaises(TypeError, math.signbit, '1.0')
+
+        for arg in [0, 0., 1, 1., INF, NAN]:
+            with self.subTest('positive', arg=arg):
+                self.assertFalse(math.signbit(arg))
+        for arg in [-0., -1, -1., NINF, NNAN]:
+            with self.subTest('negative', arg=arg):
+                self.assertTrue(math.signbit(arg))
```

Comment:
I don't think that it's relevant to test passing integers since the value is casted to a float anyway. So you can simplify tests to:

```suggestion
        for arg in [0., 1., INF, NAN]:
            with self.subTest(arg=arg):
                self.assertFalse(math.signbit(arg))
                self.assertTrue(math.signbit(-arg))
```

(No need to define `NNAN`.)